### PR TITLE
Grade genuegend

### PR DESCRIPTION
**Pull request by BojanTRB**

### CODE REVIEW

File: create_db.sql [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+CREATE TABLE Nationality (
+	Id INTEGER IDENTITY(1,1 ) PRIMARY KEY,
+	[Name] VARCHAR(50) NOT NULL,
+	UNIQUE([Name])
+)
+
+CREATE TABLE Trainer (
+	Id INTEGER IDENTITY(1,1) PRIMARY KEY,
+	[Name] VARCHAR(200) NOT NULL,
```
Comment (schletz) [inline create_db.sql:50]: Split fields.

File: create_db.sql [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	BirthDate DATE NOT NULL,
+	NationalityId INTEGER,
+	FOREIGN KEY (NationalityId) REFERENCES (Nationality.Id),
+	CHECK (Year(BirthDate) > 1950),
+	UNIQUE (Name)
+);
+
+CREATE TABLE Player (
+	Id INTEGER IDENTITY(1,1) PRIMARY KEY,
+	[Name] VARCHAR(200) NOT NULL,
```
Comment (schletz) [inline create_db.sql:60]: Split fields.

File: er_logical.puml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    * BirthDate : DATE
+    NationnalityId : INTEGER <<FK>>
+}
+
+entity Game {
+    * Id : INTEGER <<PK>> 
+    --
+    * GameDate : DATE
+    GameDay : INTEGER
+    HomeTeam : INTEGER <<FK>>
```
Comment (schletz) [inline er_logical.puml:67]: add star for NOT NULL

File: er_logical.puml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    NationnalityId : INTEGER <<FK>>
+}
+
+entity Game {
+    * Id : INTEGER <<PK>> 
+    --
+    * GameDate : DATE
+    GameDay : INTEGER
+    HomeTeam : INTEGER <<FK>> 
+    AwayTeam : INTEGER <<FK>>
```
Comment (schletz) [inline er_logical.puml:68]: add star for NOT NULL

File: create_db.sql [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+CREATE TABLE Trainer (
+	Id INTEGER IDENTITY(1,1) PRIMARY KEY,
+	[Name] VARCHAR(200) NOT NULL,
+	BirthDate DATE NOT NULL,
+	NationalityId INTEGER,
+	FOREIGN KEY (NationalityId) REFERENCES (Nationality.Id),
+	CHECK (Year(BirthDate) > 1950),
+	UNIQUE (Name)
```
Comment (schletz) [inline create_db.sql:55]: To remove

File: create_db.sql [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+);
+
+CREATE TABLE Player (
+	Id INTEGER IDENTITY(1,1) PRIMARY KEY,
+	[Name] VARCHAR(200) NOT NULL,
+	BirthDate DATE NOT NULL,
+	NationalityId INTEGER,
+	FOREIGN KEY (NationalityId) REFERENCES (Nationality.Id),
+	CHECK (Year(BirthDate) > 1950),
+	UNIQUE (Name)
```
Comment (schletz) [inline create_db.sql:65]: To remove

File: create_db.sql [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+);
+
+CREATE TABLE TeamTrainer (
+	Id INTEGER IDENTITY(1,1) PRIMARY KEY,
+	TrainerFrom DATE NOT NULL,
+	TrainerTo DATE,
+	TrainerId INTEGER,
```
Comment (schletz) [inline create_db.sql:72]: Why nullable?

File: create_db.sql [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+	TeamId INTEGER,
```
Comment (schletz) [inline create_db.sql:73]: Why nullable?

File: er_logical.puml [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    * BirthDate : DATE
+    NationalityId : INTEGER <<FK>>
+}
+
+entity TeamTrainer{
+    * Id : INTEGER <<generated>>
+    ---
+    * TrainerFrom : DATE
+    TrainerTo : DATE
+    TrainerId : INTEGER <<FK>>
```
Comment (schletz) [inline er_logical.puml:31]: Why nullable?

File: er_logical.puml [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    TeamId : INTEGER <<FK>>
```
Comment (schletz) [inline er_logical.puml:32]: Why nullable?

**Review by schletz — CHANGES_REQUESTED** [review]
Some changes are not done in all files (logical + physical + dump must be consistent). See comments in the code review. Please check for consistency after making changes.
Instead of committing multiple "fix xx.sql" you can use an amend commit, see https://github.com/schletz/Pos3xhif/blob/master/06%20Git/06_history.md#den-letzten-commit-neu-schreiben-der-amend-commit

**Comment by BojanTRB** [comment]
Amended to last commit and waiting for a new review.

**Review by schletz — APPROVED** [review]

**Pull request merged**